src/spec/server.test.js

```diff
@@ -0,0 +1,110 @@
+const request = require('supertest');
+const app = require('../server/app');
+const db = require('../../db/dbHelpers');
+
+describe('server request handling', () => {
+  beforeAll(() => db.mongoose.connect('mongodb://localhost/fecproject'));
+  afterAll(() => db.mongoose.disconnect());
+
+  test('It should respond to the GET method', () => {
+    return request(app).get('/').expect(200);
+  });
+
+  test('It should respond with 404 when given wrong URL', () => {
+    return request(app).get('/doggo').expect(404);
+  });
+
+  test('should return correct data shape for lunch menu', (done) => {
+    request(app)
+      .get('/restaurants/90976/menu/lunch')
+      .then((response) => {
+        expect(Array.isArray(response.body)).toBe(true);
+        expect(response.body.length).toBe(20);
+        done();
+      })
+      .catch(done.fail);
+  });
+
+  test('should return correct data shape for dinner menu', (done) => {
+    request(app)
+      .get('/restaurants/90976/menu/dinner')
+      .then((response) => {
+        expect(Array.isArray(response.body)).toBe(true);
+        expect(response.body.length).toBe(20);
+        done();
+      })
+      .catch(done.fail);
+  });
+
+  test('should return correct data shape for dessert menu', (done) => {
+    request(app)
+      .get('/restaurants/90976/menu/dessert')
+      .then((response) => {
+        expect(Array.isArray(response.body)).toBe(true);
+        expect(response.body.length).toBe(8);
+        done();
+      })
+      .catch(done.fail);
+  });
+
+  describe('returning only specified items based on their tags', () => {
+    test('should return only gluten-free items', (done) => {
+      request(app)
+        .get('/restaurants/90976/menu/lunch/gluten-free')
+        .then((response) => {
+          const tagObj = {};
+          response.body.forEach((item) => {
+            const tag = item.tags;
+            if (!tagObj[tag]) {
+              tagObj[tag] = 1;
+            } else {
+              tagObj[tag] += 1;
+            }
+          });
+          expect(Array.isArray(response.body)).toBe(true);
+          expect(Object.keys(tagObj).length).toBe(1);
```

Comment:
I'm not sure, but will this specifically check that the items are gluten-free?  If the server sent back a bunch of items that are all the same tag, would this still pass the test?

Reply:
The way that the router and model works is it passes through the tag found in the req.params which is passed through in the url. So with this in mind, it will only return the items that match that specific tag